### ISSUE TITLE
Enhancement: add support for includeAllProperties option

### ISSUE THREAD

**Issue by katdti** [datalust/nlog-targets-seq#32]
When serializing to a file it is possible to serialize all properties with layout render like this:

`   <layout xsi:type="JsonLayout" includeAllProperties="true" maxRecursionLimit="20" excludeProperties="...." >`

It is not possible to add includeAllProperties, maxRecursionLimit and excludeProperties to seq target.

**Background for this request**:
I am trying to log properties using nlog-targests-seq so they are searchable.
I know I can use @ to serialize objects in messages - but is seems not possible for user properties?

Pseudo c# code that works (I can e.g. search for x.status=0 i seq):
`class MyClass { public int status { get; set; } };`
`var x = new MyClass();`
`log.Debug().Message("Write {@x}", x).Write();`

Pseudo c# code not working (only x class name is serialized so I can not search for x.status=0):
`class MyClass { public int status { get; set; }`
`var x= new MyClass();`
`log.Debug().Message("Write Hi").Property("x", x).Write();`

Today I have to format every property into the message string - this does not seem like an optimal method for structured logging.

**Comment by nblumhardt** [comment]
Hi!

In the message, NLog supplies, via the message template, a hint for how a property should be captured - either as a serialized structure (`@`), or as a simple value (the default).

For attached properties, there's no hinting mechanism, so the most conservative option - capturing as a simple value.

Although it might seem like capturing structure is a reasonable alternative default, it's not always clear-cut. E.g. `Property("MessageType", message.GetType())` is a case where the value benefits enormously from being considered a string - serializing broad structures like `Type`, even to a limited depth, can generate tonnes of data very quickly.

To handle this case, Serilog (upon which NLog's new structured logging support is based) accepts a Boolean to the equivalent of `Property()`, so your example would be rewritten:

```csharp
log.Debug().Message("Write Hi").Property("x", x, capture: true).Write();
```

It'd be a great feature to look at adding to NLog; in the meantime I think the options for adding support directly here in this target are a bit limited.

Hope this helps!
Nick

**Comment by snakefoot** [comment]
@nblumhardt in the meantime I think the options for adding support directly here in this target are a bit limited.

You could provide the ability to configure MaxRecursionLimit, then it will allow advanced users to divert from the default message-template logic.

**Comment by snakefoot** [comment]
Created PR #34

**Comment by snakefoot** [comment]
@katdti Please try the latest dev-build to see if it fixes your issue: https://www.nuget.org/packages/NLog.Targets.Seq/1.0.1-dev-00047

You can now configure MaxRecursionLimit=1 and it will perform property reflection for all properties.

**Comment by katdti** [comment]
Partial success.
MaxRecursionLimit=X seems to do the expected. But
MaxRecursionLimit=1 and the following logging causes a fail:

```            
Log.Debug()
    .Message("TestMessage")
    .Property("Message", new
                    {
                        data = new int[16]
                    }
                )
    .Write();
```

With MaxRecursionLimit=2 the above works.

Callstack:
> 2018-05-29 14:37:22.7559 Error Seq(Name=seq_wrapped): Failed sending LogEvents Exception: System.Net.WebException: The remote server returned an error: (400) Bad Request.
   at System.Net.HttpWebRequest.GetResponse()
   at NLog.Targets.Seq.SeqTarget.PostBatch(IList`1 logEvents)
   at NLog.Targets.Seq.SeqTarget.Write(IList`1 logEvents)
2018-05-29 14:37:22.7559 Error AsyncWrapper(Name=seq): Error in lazy writer timer procedure. Exception: System.Net.WebException: The remote server returned an error: (400) Bad Request.
   at System.Net.HttpWebRequest.GetResponse()
   at NLog.Targets.Seq.SeqTarget.PostBatch(IList`1 logEvents)
   at NLog.Targets.Seq.SeqTarget.Write(IList`1 logEvents)
   at NLog.Targets.Target.WriteAsyncThreadSafe(IList`1 logEvents)
   at NLog.Targets.Target.WriteAsyncLogEvents(IList`1 logEvents)
   at NLog.Targets.Wrappers.AsyncTargetWrapper.WriteEventsInQueue(Int32 batchSize, String reason)
   at NLog.Targets.Wrappers.AsyncTargetWrapper.ProcessPendingEvents(Object state)

**Comment by snakefoot** [comment]
These are the different outputs:

MaxRecursionLimit=0
`{"@t":"2018-05-29T16:56:33.2727562+02:00","@l":"Info","@mt":"TestMessage","@r":null,"Message":"{ data = System.Int32[] }"}`

MaxRecursionLimit=1
`{"@t":"2018-05-29T16:57:01.4722467+02:00","@l":"Info","@mt":"TestMessage","@r":null,"Message":{"data":}}`

MaxRecursionLimit=2
`{"@t":"2018-05-29T16:57:34.5178810+02:00","@l":"Info","@mt":"TestMessage","@r":null,"Message":{"data":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]}}`

**Comment by snakefoot** [comment]
@katdti Have created https://github.com/NLog/NLog/pull/2747 that will resolve the issue with `MaxRecursionLimit=1` (Hopefully it will be included in the soon to come NLog 4.5.6)

**Comment by snakefoot** [comment]
@katdti  Please try with the just released NLog 4.5.6: https://www.nuget.org/packages/NLog

**Comment by katdti** [comment]
Excellent work. Works perfect now.
